pycbc/inference/sampler_kombine.py

```diff
@@ -0,0 +1,194 @@
+        """
+        blob0 = None
+        if self.burn_in_iterations == 0:
+            # no burn in, use the initial positions
+            p0 = self.p0
+            if self.likelihood_evaluator.return_meta:
+                blob0 = [self.likelihood_evaluator(p0[wi,:])[1]
+                    for wi in range(self.nwalkers)]
+        else:
+            p0 = None
+            # kombine requires blob data to be specified
+            if self.likelihood_evaluator.return_meta:
+                blob0 = self._sampler.blobs[-1]
+        kwargs['blob0'] = blob0
+        res = self._sampler.run_mcmc(niterations, p0=p0, **kwargs)
+        p, lnpost, lnprop = res[0], res[1], res[2] 
+        # update the positions
+        self._pos = p
+        return p, lnpost, lnprop
+
+    @property
+    def lnpost(self):
+        """ Get the natural logarithm of the likelihood as an 
+        nwalkers x niterations array.
+        """
+        # kombine returns niterations x nwaklers
+        return self._sampler.lnpost.transpose()
+
+    @property
+    def chain(self):
+        """Get all past samples as an nwalker x niterations x ndim array."""
+        # kombine returns niterations x nwalkers x ndim
+        return self._sampler.chain.transpose((1,0,2))
+
+    def burn_in(self):
+        """Use kombine's `burnin` routine to advance the sampler.
+        
+        If a minimum number of burn-in iterations was specified, this will run
+        the burn-in until it has advanced at least as many steps as desired.
+        The initial positions (p0) must be set prior to running.
+
+        For more details, see `kombine.sampler.burnin`.
+
+        Returns
+        -------
+        p : numpy.array
+            An array of current walker positions with shape (nwalkers, ndim).
+        lnpost : numpy.array
+            The list of log posterior probabilities for the walkers at
+            positions p, with shape (nwalkers, ndim).
+        lnprop : numpy.array
+            The list of log proposal densities for the walkers at positions p,
+            with shape (nwalkers, ndim).
+        """
+        # check that we haven't already burned in
+        if self.pos is not None:
+            raise ValueError("burn-in already run")
+        # run once
+        p0 = self.p0
+        if self.likelihood_evaluator.return_meta:
+            blob0 = [self.likelihood_evaluator(p0[wi,:])[1]
+                for wi in range(self.nwalkers)]
+        else:
+            blob0 = None
+        res = self._sampler.burnin(self.p0, blob0=blob0)
+        p, post, q = res[0], res[1], res[2]
+        # continue running until minimum burn in is satistfied
+        while self.niterations < self.burn_in_iterations:
+            p0 = p
+            res = self._sampler.burnin(p0)
+            p, post, q = res[0], res[1], res[2]
+            # update position
+            self._pos = p
+        self.burn_in_iterations = self.niterations
+        return p, post, q
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->